clevertap/src/main/java/com/rudderstack/android/integrations/clevertap/CleverTapIntegrationFactory.java

```diff
@@ -1,47 +1,398 @@
+                                    @NonNull Activity activity,
+                                    @Nullable Bundle bundle
+                            ) {
+                                // Nothing to implement
+                            }
+
+                            @Override
+                            public void onActivityDestroyed(@NonNull Activity activity) {
+                                // Nothing to implement
+                            }
+
+                            @SuppressLint("RestrictedApi")
+                            @Override
+                            public void onActivityStarted(@NonNull Activity activity) {
+                                // Nothing to implement
+                            }
+                        }
+                );
+    }
+
+    //  handling life cycle methods of an Application
+
+    private void processRudderEvent(RudderMessage element) {
+        String type = element.getType();
+        if (type != null) {
+            switch (type) {
+                case MessageType.IDENTIFY:
+                    String userId = element.getUserId();
+                    if (!TextUtils.isEmpty(userId)) {
+                        Map<String, Object> traits = transformTraits(element.getTraits());
+                        try {
+                            cleverTap.onUserLogin(traits);
+                        } catch (Exception e) {
+                            e.printStackTrace();
+                        }
+                    }
+                    break;
+                case MessageType.TRACK:
+                    String eventName = element.getEventName();
+                    try {
+                        if (eventName != null) {
+                            Map<String, Object> eventProperties = element.getProperties();
+                            if (
+                                    eventName.equals("Order Completed") && eventProperties != null
+                            ) {
+                                handleEcommerceEvent(eventProperties);
+                                return;
+                            }
+                            if (eventProperties == null || eventProperties.size() == 0) {
+                                cleverTap.pushEvent(eventName);
+                                return;
+                            }
+                            cleverTap.pushEvent(eventName, eventProperties);
+                        }
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                    }
+                    break;
+                case MessageType.SCREEN:
+                    String screenName = element.getEventName();
+                    Map<String, Object> screenProperties = element.getProperties();
+                    if (screenProperties != null) {
+                        cleverTap.pushEvent(
+                                String.format("Viewed %s screen", screenName),
```

Comment:
if screen name not present Viewed double space screen will be printed if no issue keep it this way otherwise check if screenName is present or not

Reply:
Since we are retrieving `screenName` by calling `element.getEventName()` which is always set and non-null in the case of android sdk, we are safe

Reply:
got it...